lib/evmone/eof.cpp

```diff
@@ -576,9 +625,34 @@ EOF1Header read_valid_eof1_header(bytes_view container)
     header.data_size =
         section_headers[DATA_SECTION].empty() ? uint16_t{0} : section_headers[DATA_SECTION][0];
 
+    header.container_sizes = section_headers[CONTAINER_SECTION];
+    std::vector<uint16_t> container_offsets;
+    auto container_offset = code_offset + header.data_size;
+    for (const auto container_size : header.container_sizes)
+    {
+        header.container_offsets.emplace_back(static_cast<uint16_t>(container_offset));
+        container_offset += container_size;
+    }
+
     return header;
 }
 
+void append_data_section(bytes& container, bytes_view aux_data)
+{
+    const auto header = read_valid_eof1_header(container);
+    const auto insert_pos =
+        header.container_offsets.empty() ? container.size() : header.container_begin(0);
+    container.insert(insert_pos, aux_data);
+    const auto new_data_size = header.data_size + aux_data.size();
+
```

Comment:
Should check/assert for `new_data_size < 2^16`.

Reply:
Added the check.